components/director/internal/domain/runtime/service.go

```diff
@@ -337,7 +360,39 @@ func (s *service) ListLabels(ctx context.Context, runtimeID string) (map[string]
 		return nil, errors.Wrap(err, "while getting label for Runtime")
 	}
 
-	return labels, nil
+	return s.extractNotProtectedLabels(labels, s.protectedLabelPattern)
+}
+
+func (s *service) extractNotProtectedLabels(labels map[string]*model.Label, protectedLabelsKeyPattern string) (map[string]*model.Label, error) {
```

Comment:
I know it's a good practice in most places, especially in the Java world to place dependant methods beneaf each other, but can we move the `extractNotProtectedLabels`, `unsafeExtractNotProtectedLabels` and `isProtected` utility functions below the `DeleteLabel` function? In most of this project I see that the pattern is to have all exported methods first and right after them are the utility/unexported methods. So just for consistency sake I was thinking along those lines - what do you think?

Reply:
Also maybe - `extractUnprotectedLabels` as name?

Reply:
I agree, I didn't put them like this for any particular reason, just didn't move them below while coding in order to be able to see the code that I write. I will move them now.